langmeta/shared/src/main/scala/org/langmeta/semanticdb/ResolvedSymbol.scala

```diff
@@ -1,9 +1,65 @@
 package org.langmeta
 package semanticdb
 
-final case class ResolvedSymbol(symbol: Symbol, denotation: Denotation) {
+import scala.runtime.AbstractFunction2
+
+final case class ResolvedSymbol2(symbol: Symbol, denotation: Denotation, members: List[Signature]) {
   def input: Input = Input.Denotation(denotation.signature, symbol)
-  def syntax = s"${symbol.syntax} => ${denotation.syntax}"
   def structure = s"""ResolvedSymbol(${symbol.structure}, ${denotation.structure})"""
-  override def toString = syntax
+}
+
+final class ResolvedSymbol(
+    val symbol: Symbol,
+    val denotation: Denotation,
+    val members: List[Signature])
+    extends Product
+    with Serializable
+{
+  def this(symbol: Symbol, denotation: Denotation) = this(symbol, denotation, Nil)
+  def input: Input = Input.Denotation(denotation.signature, symbol)
+  def syntax: String = {
+    val memberSyntax = if (members.isEmpty) "" else s".{+${members.length}}"
+    s"${symbol.syntax} => ${denotation.syntax}$memberSyntax"
+  }
+  def structure: String = {
+    val memberSyntax = if (members.isEmpty) "" else s", {+${members.length}}"
+    s"""ResolvedSymbol(${symbol.structure}, ${denotation.structure}$memberSyntax)"""
+  }
+  // We must implement the boilerplate below in order to keep binary compatibility
+  // with previous releases.
```

Comment:
Is `copy` the only problem, or there is something else?

Reply:
Also everything interited from Product + companion object apply/unapply + object extends FunctionN